app/containers/ChatContainer/reducer.js

```diff
@@ -14,15 +14,14 @@ const initialState = fromJS({
   messages: [],
 });
 
-function chatContainerReducer(state = initialState, action) {
-  switch (action.type) {
+function chatContainerReducer(state = initialState, { type, payload }) {
+  switch (type) {
     case SEND_MESSAGE_ACTION:
       return state;
-    case RECEIVE_MESSAGE_ACTION:
-      return {
-        messages: state.messages.push(action.message),
-        ...state,
-      };
+    case RECEIVE_MESSAGE_ACTION: {
+      const messages = state.get('messages');
+      return state.set('messages', messages.push(payload));
```

Comment:
`message.push(...)` creates a new messages object? we defined a `const` earlier so cannot mutate `messages` object itself

Reply:
a) Actually `const` only means that you cannot redeclare it (`const a = 1; const a = 2`), you are free to mutate its state however you want
b) This is an Immutable List (https://facebook.github.io/immutable-js/docs/#/List/push), so pushing returns a copy here
c) I changed to `update` because it does this work in 1 line.

Reply:
oh jesus *pwned*

thank god im not interviewing for frontend eng positions